internal/cri/server/images/image_pull.go

```diff
@@ -334,11 +334,30 @@ func (c *CRIImageService) createImageReference(ctx context.Context, name string,
 	} else if !errdefs.IsAlreadyExists(err) {
 		return err
 	}
-	if oldImg.Target.Digest == img.Target.Digest && oldImg.Labels[crilabels.ImageLabelKey] == labels[crilabels.ImageLabelKey] {
+	// Retrieve oldImg from image store here because Create routine returns an
+	// empty image on ErrAlreadyExists
+	oldImg, err := c.images.Get(ctx, name)
+	if err != nil {
+		return err
+	}
+	var fieldpaths []string
+	if oldImg.Target.Digest != img.Target.Digest {
```

Comment:
if the target digests don't match not sure we should continue on

the removed line was checking if the digests are the same and old image label key has the generated label key pinned or not?  then return fast.. no update needed

here I think we are just initializing field paths 0 to have "target" in case one of the two below require update...